apps/remote_control/lib/lexical/remote_control/code_intelligence/ast.ex

```diff
@@ -0,0 +1,77 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.Ast do
+  alias Lexical.Document
+  alias Lexical.Document.Position
+  alias Lexical.RemoteControl.CodeMod.Ast.Aliases
+
+  require Logger
+
+  @doc """
+  Expands the aliases in the given `document`, `postion` and `module_aliases`.
+
+  When we refer to a module, it's usually a short name,
+  so it's probably aliased or in a nested module,
+  so we need to find the real full name of the module at the cursor position.
+
+  For example, if we have:
+
+    ```elixir
+    defmodule Project do
+      defmodule Issue do
+        defstruct [:message]
+      end
+
+      def message(%Issue{|} = issue) do # cursor marked as `|`
+      end
+    end
+    ```
+
+  Then the the expaned module is `Project.Issue`.
+
+  Another example:
+
+    ```elixir
+    defmodule Project do
+      defmodule Issue do
+        defstruct [:message]
+      end
+    end
+
+    defmodule MyModule do
+      alias Project, as: MyProject
+
+      def message(%MyProject.Issue{|} = issue) do
+      end
+    end
+    ```
+
+  Then the the expaned module is still `Project.Issue`.
+
+  And sometimes we can't find the full name by the `Aliases.at/2` function,
```

Comment:
why?

Reply:
For example, without alias, like `%Foo.Structs.Struct{|}`